web/frontend/src/components/keyboard-input/index.svelte

```diff
@@ -0,0 +1,145 @@
+<script lang="ts">
+  /* eslint-disable id-length */
+  import { onDestroy } from 'svelte';
+
+  export type Keys = 'w' | 'a' | 's' | 'd';
+
+  import {
+    mdiArrowUp as w,
+    mdiRestore as a,
+    mdiReload as d,
+    mdiArrowDown as s,
+  } from '@mdi/js';
+  import Icon from '../icon/index.svelte';
+
+  export let isActive: boolean;
+  export let onKeyDown: (key: Keys) => void;
+  export let onKeyUp: (key: Keys) => void;
+  export let onUpdateKeyboardState: (isActive: boolean) => void;
+
+  const keyIcons = { w, a, s, d };
+
+  const pressedKeys = {
+    w: false,
+    a: false,
+    s: false,
+    d: false,
+  };
+
+  const keysLayout = [['a'], ['w', 's'], ['d']] as const;
+
+  const normalizeKey = (key: string): Keys | null => {
+    return (
+      (
+        {
+          w: 'w',
+          a: 'a',
+          s: 's',
+          d: 'd',
+          arrowup: 'w',
+          arrowleft: 'a',
+          arrowdown: 's',
+          arrowright: 'd',
+        } as Record<string, Keys>
+      )[key.toLowerCase()] ?? null
+    );
+  };
+
+  const dispatchKeyDown = (key: Keys) => {
+    pressedKeys[key] = true;
+    onKeyDown(key);
+  };
+
+  const dispatchKeyUp = (key: Keys) => {
+    pressedKeys[key] = false;
+    onKeyUp(key);
+  };
+
+  const handleKeyDown = (event: KeyboardEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const key = normalizeKey(event.key);
+
+    if (key === null || pressedKeys[key]) {
+      return;
+    }
+
+    dispatchKeyDown(key);
+  };
+
+  const handleKeyUp = (event: KeyboardEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const key = normalizeKey(event.key);
+    if (key !== null) {
+      dispatchKeyUp(key);
+    }
+  };
+
+  const toggleKeyboard = (nowActive: boolean) => {
+    if (nowActive) {
+      window.addEventListener('keydown', handleKeyDown, false);
+      window.addEventListener('keyup', handleKeyUp, false);
+    } else {
+      window.removeEventListener('keydown', handleKeyDown);
+      window.removeEventListener('keyup', handleKeyUp);
+    }
+
+    onUpdateKeyboardState(nowActive);
+  };
+
+  const handlePointerDown = (key: Keys) => {
+    dispatchKeyDown(key);
+  };
+
+  const handlePointerUp = (key: Keys) => {
+    dispatchKeyUp(key);
+  };
+
+  $: if (!isActive) {
+    toggleKeyboard(false);
+  }
+
+  onDestroy(() => {
+    toggleKeyboard(false);
+  });
+</script>
+
+<div>
+  <v-switch
+    label={isActive ? 'Keyboard enabled' : 'Keyboard disabled'}
```

Comment:
We switched the language here during design maintenance last week to just be static:
```suggestion:
    label='Keyboard enabled'

Reply:
Or maybe I just changed the other switches and forgot this one 🙈

Reply:
I believe you [left this one](https://github.com/viamrobotics/rdk/pull/2489/files#diff-5261e8e828505eac30985c4dd6dd932cd6bf18e698d1a02a686275045ca0f58c) because I was in the midst of the Svelte conversion.

What do we want the UX to be? Currently there's a toggle that the user must first enable.

Reply:
Ah yes! Thanks for refreshing my memory, nvm then, we can hold this off until the next maintenance. We're just ensuring that all switches have unchanging text that follows the format [verb] [noun]. So for example in camera it's `View camera`. So I guess here the change would be `Enable keyboard`